runtime/near-vm-logic/src/logic.rs

```diff
@@ -115,8 +113,11 @@ impl<'a> VMLogic<'a> {
         // Overflow should be checked before calling VMLogic.
         let current_account_balance = context.account_balance + context.attached_deposit;
         let current_storage_usage = context.storage_usage;
-        let max_gas_burnt = if context.is_view {
-            config.limit_config.max_gas_burnt_view
+        let max_gas_burnt = if context.view_config.is_view {
+            match context.view_config.max_gas_burnt_view {
+                Some(max_gas_burnt_view) => max_gas_burnt_view,
+                None => config.limit_config.max_gas_burnt,
```

Comment:
Hm, am I correct that this is essentially dead code? Every time we view_config.is_view is true, view_cconfig.max_gas_burnt is Some?

If that's the case, I suggest making this code staticall unreachable, as follows: 

```
impl VmContext {
  view_config: Option<ViewConfig>
}

struct ViewConfig {
  max_gas_burnt: gas
}
```

than, we'll be passing `Option<VIewConfig>` everywhere, with the semantics that we are in a view if the config is Some.